upup/pkg/fi/cloudup/vspheretasks/attachiso.go

```diff
@@ -17,19 +17,56 @@ limitations under the License.
 package vspheretasks
 
 import (
+	"bytes"
+	"fmt"
 	"github.com/golang/glog"
+	"github.com/pborman/uuid"
+	"io/ioutil"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/pkg/model"
 	"k8s.io/kops/upup/pkg/fi"
 	"k8s.io/kops/upup/pkg/fi/cloudup/vsphere"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
 )
 
 // AttachISO represents the cloud-init ISO file attached to a VMware VM
 //go:generate fitask -type=AttachISO
 type AttachISO struct {
-	Name *string
-	VM   *VirtualMachine
```

Comment:
I believe I have found the problem that you are talking about. This is the code responsible for ignoring or accounting a dependency for reflect.Struct https://github.com/vmware/kops/blob/master/upup/pkg/fi/topological_sort.go#L98 

The problem is that current code identifies only those structs that are listed in the if-else structure. The structs that you are adding to your task, like InstanceGroup and BootstratScript don't end up getting processed by last else-block and hence the error "Unhandled type for...".

This if-else structure is not the ideal way of defining dependencies. But implementing HasDepenency method is also bit of a redundant work.

I would recommend maintaining a set of types, like InstanceGroup and BootScript, which we don't want to include as a dependency. Check that in the above discussed if-else block.